### PR TITLE
refactor(osd): implement container inspector for a single container

### PR DESCRIPTION
**Pull request by smira**
Instead of pulling a full list of containers, implement inspector query
for a single container following the spec to build display name.

Also adds many more tests for the container inspector.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Comment by smira** [comment]
WIP, something is broken around `logs -k`, but I haven't found the root cause yet

File: internal/pkg/containers/containers_test.go [hunk 1]
```diff
@@ -42,6 +44,25 @@ type ContainersSuite struct {
...
+
+	containerRunners []runner.Runner
+	containersWg     sync.WaitGroup
+}
+
+// WithAnnotations appends or replaces the annotations on the spec with the
+// provided annotations
+//
+// TODO: taken from containerd/oci > 1.2.6
+func WithAnnotations(annotations map[string]string) oci.SpecOpts {
```
Comment (andrewrynhard) [inline internal/pkg/containers/containers_test.go]: 👍

File: internal/app/osd/internal/reg/reg.go [hunk 2]
```diff
@@ -289,28 +284,27 @@ func (r *Registrator) Dmesg(ctx context.Context, in *empty.Empty) (data *proto.D
...
 	case req.Namespace == "system" || req.Id == "kubelet" || req.Id == "kubeadm":
-		filename = filepath.Join("/var/log", filepath.Base(req.Id)+".log")
+		filename := filepath.Join("/var/log", filepath.Base(req.Id)+".log")
+		var file *os.File
+		file, err = os.OpenFile(filename, os.O_RDONLY, 0)
+		if err != nil {
+			return
+		}
+		// nolint: errcheck
+		defer file.Close()
```
Comment (smira) [inline internal/app/osd/internal/reg/reg.go]: I believe this was missing in the previous implementation which should have lead to fd leak

**Comment by smira** [comment]
I've got logs fixed, now new code works exactly like the old one, but also there's a code to catch up and read the live log which usually isn't triggered if sandbox/pod is found, but we might want to activate it in the future as the last resort.

**Comment by smira** [comment]
tl;dr version: when requesting info for a single container, don't list all the containers

I tried my best to do matching between pods/containers more robust, but my end goal is try out CRI next for k8s namespace. 

We still can't get logs for k8s containers reliable as we can't pull restartCount via containerd API
